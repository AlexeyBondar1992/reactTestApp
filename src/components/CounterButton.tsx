import React, { Component } from 'react';

interface ICounterButtonProps {
    color: string;
}

interface ICounterButtonState {
    count: number;
}

class CounterButton extends Component<ICounterButtonProps, ICounterButtonState> {
    constructor (props: ICounterButtonProps) {
        super(props);
        this.state = {
            count: 0
        };
    }

    shouldComponentUpdate (nextProps: ICounterButtonProps, nextState: ICounterButtonState, nextContext: any) {
        return nextState.count !== this.state.count;
    }

    updateCount = () => {
        this.setState(state => ({count: state.count + 1}))
    };

    render () {
        return (
            <button color={this.props.color} onClick={this.updateCount} id="counter">
                Count: {this.state.count}
            </button>
        );
    }
}

export default CounterButton;