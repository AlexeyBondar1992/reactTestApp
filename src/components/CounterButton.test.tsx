import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import CounterButton from './CounterButton';

const mockColor = 'red';

describe('CounterButton', () => {
    it('should test snapshot to match', () => {
        expect(toJson(shallow(<CounterButton color={mockColor}/>))).toMatchSnapshot();
    });

    it('correctly increments the counter', () => {
        const wrapper = shallow(<CounterButton color={mockColor}/>);

        wrapper.find('[id="counter"]').simulate('click');
        expect(wrapper.state()).toEqual({ count: 1});

        wrapper.find('[id="counter"]').simulate('click');
        expect(wrapper.state()).toEqual({ count: 2});

        expect(wrapper.props().color).toEqual(mockColor);
    });
});
