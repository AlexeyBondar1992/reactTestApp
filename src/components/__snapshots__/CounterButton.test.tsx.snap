// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton should test snapshot to match 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
